compiler/rustc_trait_selection/src/traits/error_reporting/ambiguity.rs

```diff
@@ -52,10 +53,21 @@ pub fn compute_applicable_impls_for_diagnostics<'tcx>(
                 _ => return false,
             }
 
-            let impl_predicates = tcx.predicates_of(impl_def_id).instantiate(tcx, impl_args);
-            ocx.register_obligations(impl_predicates.predicates.iter().map(|&predicate| {
-                Obligation::new(tcx, ObligationCause::dummy(), param_env, predicate)
-            }));
+            let obligations = tcx
+                .predicates_of(impl_def_id)
+                .instantiate(tcx, impl_args)
+                .into_iter()
+                .map(|(predicate, _)| {
+                    Obligation::new(tcx, ObligationCause::dummy(), param_env, predicate)
+                })
+                // Kinda hacky, but let's just throw away obligations that overflow.
```

Comment:
I was wondering if we should add a FIXME here or something and if you have some kind of thought about how we could potentially get rid of the "hack" at some point.

Reply:
Yeah, at that point it will be obvious but maybe worth adding a `FIXME(next_solver)` or something like that